lib/msal-common/src/auth/IdToken.ts

```diff
@@ -0,0 +1,123 @@
+ * Licensed under the MIT License.
+ */
+
+import { ClientAuthError } from "../error/ClientAuthError";
+import { StringDict } from "../utils/MsalTypes";
+import { StringUtils } from "../utils/StringUtils";
+import { ICrypto } from "../utils/crypto/ICrypto";
+
+/**
+ * @hidden
+ */
+export class IdToken {
+
+    issuer: string;
+    objectId: string;
+    subject: string;
+    tenantId: string;
+    version: string;
+    preferredName: string;
+    name: string;
+    homeObjectId: string;
+    nonce: string;
+    expiration: string;
+    rawIdToken: string;
+    claims: StringDict;
+    sid: string;
+    cloudInstance: string;
+    /* tslint:disable:no-string-literal */
+    constructor(rawIdToken: string, crypto: ICrypto) {
+        if (StringUtils.isEmpty(rawIdToken)) {
+            throw ClientAuthError.createIdTokenNullOrEmptyError(rawIdToken);
+        }
+        try {
+            this.rawIdToken = rawIdToken;
+            this.claims = IdToken.extractIdToken(rawIdToken, crypto);
+            if (this.claims) {
+                if (this.claims.hasOwnProperty("iss")) {
+                    this.issuer = this.claims["iss"];
+                }
+
+                if (this.claims.hasOwnProperty("oid")) {
+                    this.objectId = this.claims["oid"];
+                }
+
+                if (this.claims.hasOwnProperty("sub")) {
+                    this.subject = this.claims["sub"];
+                }
+
+                if (this.claims.hasOwnProperty("tid")) {
+                    this.tenantId = this.claims["tid"];
+                }
+
+                if (this.claims.hasOwnProperty("ver")) {
+                    this.version = this.claims["ver"];
+                }
+
+                if (this.claims.hasOwnProperty("preferred_username")) {
+                    this.preferredName = this.claims["preferred_username"];
+                }
+
+                if (this.claims.hasOwnProperty("name")) {
+                    this.name = this.claims["name"];
+                }
+
+                if (this.claims.hasOwnProperty("nonce")) {
+                    this.nonce = this.claims["nonce"];
+                }
+
+                if (this.claims.hasOwnProperty("exp")) {
+                    this.expiration = this.claims["exp"];
+                }
+
+                if (this.claims.hasOwnProperty("home_oid")) {
+                    this.homeObjectId = this.claims["home_oid"];
+                }
+
+                if (this.claims.hasOwnProperty("sid")) {
+                    this.sid = this.claims["sid"];
+                }
+
+                if (this.claims.hasOwnProperty("cloud_instance_host_name")) {
+                    this.cloudInstance = this.claims["cloud_instance_host_name"];
+                }
+                /* tslint:enable:no-string-literal */
+            }
+        } catch (e) {
+            /*
+             * TODO: This error here won't really every be thrown, since extractIdToken() returns null if the decodeJwt() fails.
```

Comment:
if this never throws should we not use a try catch here/

Reply:
Agreed, wanted to revisit when we added functionality code but I can update it here.

Reply:
+1

Reply:
Removed try/catch.